src/ranged.cpp

```diff
@@ -948,43 +943,42 @@ static void do_aim( player *p, std::vector <Creature *> &t, int &target,
         p->recoil = std::max(MIN_RECOIL, p->recoil);
     }
 
-    const int aim_amount = p->aim_per_time( *relevant, p->recoil );
+    const double aim_amount = p->aim_per_move( *relevant, p->recoil );
     if( aim_amount > 0 ) {
         // Increase aim at the cost of moves
-        p->moves -= 10;
+        p->moves--;
```

Comment:
If profiling later showed a bottleneck could precalculate `improv` for all common values of `cost`
```
# Character::aim_per_move
double improv = 1.0 - pow( 0.5, 1.0 / ( cost * k ) );
```

That said though I'm running with no optimizations `RELEASE=0` on an ancient laptop and there isn't a performance concern so I'm very reluctant to start blind optimization without any evidence the current code is too slow.

Note also that loop runs millions of times during the stats dumper without any obvious slowdown.